packages/hardhat-core/src/types/config.ts

```diff
@@ -170,11 +171,22 @@ export interface HttpNetworkHDAccountsConfig {
 export interface HardhatNetworkMiningConfig {
   auto: boolean;
   interval: number | [number, number];
+  mempool: HardhatNetworkMempoolConfig;
 }
 
 export interface HardhatNetworkMiningUserConfig {
   auto?: boolean;
   interval?: number | [number, number];
+  mempool?: HardhatNetworkMempoolUserConfig;
+}
+
+export interface HardhatNetworkMempoolConfig {
+  // order should be on of `HARDHAT_MEMPOOL_SUPPORTED_ORDERS` entries
+  order: typeof HARDHAT_MEMPOOL_SUPPORTED_ORDERS[number];
```

Comment:
I'm going to change this to `string`. We recently [had a similar conversation](https://github.com/nomiclabs/hardhat/pull/1928/files/199053ebb11514426e8a4d152bf33c680ca1cd99#r746783321) where we decided to use a slightly unsafe to avoid exposing something from `internal`.

Also, i think having `string` here makes me feel more comfortable with the possibility of adding a new option in the future. The reason is that this string would be guaranteed at runtime to have a valid value, so expanding the set of values won't be a compilation error.